src/diagnostics/land_compute_methods.jl

```diff
@@ -0,0 +1,146 @@
+@land_compute "medlyn_term" SoilCanopyModel p.canopy.conductance.medlyn_term
+@land_compute "stomatal_conductance" SoilCanopyModel p.canopy.conductance.gs
+
+# Canopy - Energy
+@land_compute "canopy_aerodynamic_resistance" SoilCanopyModel p.canopy.energy.r_ae
+@land_compute "canopy_latent_heat_flux" SoilCanopyModel p.canopy.energy.lhf
+@land_compute "canopy_sensible_heat_flux" SoilCanopyModel p.canopy.energy.shf
+
+# Canopy - Hydraulics
+@land_compute "area_index" SoilCanopyModel p.canopy.hydraulics.area_index.leaf
+@land_compute "canopy_water_potential" SoilCanopyModel p.canopy.hydraulics.ψ
+@land_compute "cross_section" SoilCanopyModel p.canopy.hydraulics
+@land_compute "cross_section_roots" SoilCanopyModel p.canopy.hydraulics.fa_roots
+@land_compute "moisture_stress_factor" SoilCanopyModel p.canopy.hydraulics.β
+
+# Canopy - Photosynthesis
+@land_compute "photosynthesis_net_canopy" SoilCanopyModel p.canopy.photosynthesis.GPP
+@land_compute "photosynthesis_net_leaf" SoilCanopyModel p.canopy.photosynthesis.An
+@land_compute "respiration_leaf" SoilCanopyModel p.canopy.photosynthesis.Rd
+@land_compute "vcmax25" SoilCanopyModel p.canopy.photosynthesis.vcmax25
+
+# Canopy - Radiative Transfer
+@land_compute "near_infrared_radiation" SoilCanopyModel p.canopy.radiative_transfer.nir
+@land_compute "near_infrared_radiation_absorbed" SoilCanopyModel p.canopy.radiative_transfer.anir
+@land_compute "near_infrared_radiation_reflected" SoilCanopyModel p.canopy.radiative_transfer.rnir
+@land_compute "near_infrared_radiation_transmitted" SoilCanopyModel p.canopy.radiative_transfer.tnir
+@land_compute "photosynthetically_active_radiation" SoilCanopyModel p.canopy.radiative_transfer.par
+@land_compute "photosynthetically_active_radiation_absorbed" SoilCanopyModel p.canopy.radiative_transfer.apar
+@land_compute "photosynthetically_active_radiation_reflected" SoilCanopyModel p.canopy.radiative_transfer.rpar
+@land_compute "photosynthetically_active_radiation_transmitted" SoilCanopyModel p.canopy.radiative_transfer.tpar
+@land_compute "radiation_longwave_net" SoilCanopyModel p.canopy.radiative_transfer.LW_n
+@land_compute "radiation_shortwave_net" SoilCanopyModel p.canopy.radiative_transfer.SW_n
+
+# Canopy - Transpiration
+@land_compute "canopy_transpiration" SoilCanopyModel p.canopy.transpiration
+
+## Drivers Module ##
+
+@land_compute "air_co2" SoilCanopyModel p.drivers.c_co2
+@land_compute "pressure" SoilCanopyModel p.drivers.P
+@land_compute "rainfall" SoilCanopyModel p.drivers.P_liq
+@land_compute "radiation_longwave_down" SoilCanopyModel p.drivers.LW_d
+@land_compute "radiation_shortwave_down" SoilCanopyModel p.drivers.SW_d
+@land_compute "snowfall" SoilCanopyModel p.drivers.P_snow
+@land_compute "solar_zenith_angle" SoilCanopyModel p.drivers.θs
+@land_compute "specific_humidity" SoilCanopyModel p.drivers.q
+@land_compute "wind_speed" SoilCanopyModel p.drivers.u
+
+## Soil Module ##
+
+@land_compute "infiltration" SoilCanopyModel p.soil.infiltration
```

Comment:
also here a thought: do we want namespace or keep directly p.soil.X

Reply:
For soil, we just have p.soil.Var, and not p.soil.energy.Var or p.soil.water.Var. 

I think this makes sense because we dont currently have different "components" of the soil model which require modularity in the same way that we do for the canopy, but we could have designed that. it might be nice at some point (components could be water, energy, CO2, ...).

Reply:
Right! 
No need to make any change until it becomes useful - but let's keep it in mind =)

Reply:
💯 !!